constrained_linear_regression/__init__.py

```diff
@@ -78,6 +84,195 @@ def fit(self, X, y, min_coef=None, max_coef=None):
         return self
 
 
+class MultiConstrainedLinearRegression(ConstrainedLinearRegression):
+    """
+    This class implementation extends the ConstrainedLinearRegression class to handle multiple constraints.
+
+    Parameters
+    ----------
+    fit_intercept : bool, default=True
+        Whether or not to calculate the intercept for this model.
+
+    normalize : bool, default=False
+        This parameter is ignored when fit_intercept is set to False. 
+        If True, the regressors X will be normalized before regression by subtracting the mean and dividing by the l2-norm.
+
+    copy_X : bool, default=True
+        If True, X will be copied; else, it may be overwritten.
+
+    nonnegative : bool, default=False
+        When set to True, forces the coefficients to be positive.
+
+    ridge : float, default=0
+        When set to a float value, adds RIDGE regularization to the model, with the supplied alpha.
+
+    lasso : float, default=0
+        When set to a float value, adds LASSO regularization to the model, with the supplied alpha.
+
+    tol : float, default=1e-15
+        The tolerance for the optimization.
+
+    learning_rate : float, default=1.0
+        The learning rate for the optimization.
+
+    max_iter : int, default=10000
+        The maximum number of iterations for the optimization algorithm.
+
+    Methods
+    --------
+    fit(X, y, horizon, min_coef=None, max_coef=None, initial_beta=None):
+        Fits the model according to the given training data.
+    """
+    global_horizon_count = 0
+    
+    def __init__(
+            self,
+            fit_intercept=True,
+            normalize=False,
+            copy_X=True,
+            nonnegative=False,
+            ridge=0,
+            lasso=0,
+            tol=1e-15,
+            learning_rate=1.0,
+            max_iter=10000,
+            penalty_rate=0
+    ):
+        super().__init__(fit_intercept, normalize, copy_X, nonnegative, ridge, lasso, tol, learning_rate, max_iter)
+        self.penalty_rate = penalty_rate
+
+    def fit(self, X, y, horizon, min_coef=None, max_coef=None, initial_beta=None):
```

Comment:
Can you please write a unit test for this new class in the `tests` folder?

Reply:
Sure! [7f03cf9](https://github.com/avidale/constrained-linear-regression/pull/7/commits/7f03cf9598b2c7b11cd48f4ee67352eef0f7f2b0)